server/http.go

```diff
@@ -401,55 +391,41 @@ func (p *Plugin) handleStartMeeting(w http.ResponseWriter, r *http.Request) {
 	}
 
 	meetingID := zoomUser.Pmi
-
-	err = p.postMeeting(user, meetingID, req.ChannelID, req.Topic)
-	if err != nil {
+	if err = p.postMeeting(user, meetingID, req.ChannelID, req.Topic); err != nil {
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
 
-	meetingURL := p.getMeetingURL(meetingID, userID)
-
+	meetingURL := p.getMeetingURL(user, meetingID, req.ChannelID)
```

Comment:
@larkox After some digging around, I began thinking that we can just get rid of including `channelID` from the OAuth flow altogether, so that's how I solved the above issue in this commit: https://github.com/mattermost/mattermost-plugin-zoom/pull/150/commits/163a7ed0502d81bd962c33a40c3624deffc157a6

Please also see https://github.com/mattermost/mattermost-plugin-zoom/pull/150#discussion_r462435568 and let me know if I'm missing something.

Reply:
User state needs the channel ID so when the OAuth flow finishes and calls oauthComplete, it starts the meeting in the right channel. You removed the channel ID from the connect part, but you are still considering the channelID exist in the complete part. This should throw a panic of index out of bounds when you try to connect.

Maybe this is a problem that cannot be solved, and the channelID has to be passed this way, but it would be great that we could find a better solution.

Reply:
@larkox Looks like I totally missed the `channelID` usage in the OAuth completion flow, thanks for pointing that out. 

> This should throw a panic of index out of bounds when you try to connect.

I can't test the flow fully because I get this message from Zoom when I try to complete the auth flow:
```
Invalid redirect: http://XXXXXXXXXXXX.ngrok.io/plugins/zoom/oauth2/complete (4,700)
```

This also happens on the master branch, so I'm guessing that Zoom doesn't work without HTTPS or I'm missing a configuration setting.

---

Anyway, before I give up on trying to find a "clean" solution to the issue at hand, here's my last attempt: https://github.com/mattermost/mattermost-plugin-zoom/pull/150/commits/8dec813ce17a3e8cd3255864d03469b8aba996b4 😄 

The idea is to build the state without `channelID`, but store `channelID` as the value along with the state in the KV store. 

We achieve this by saving the state inside the `/zoom start` command handler if an authenticated user cannot be found:
 * We can still access `channelID` from the KV store given the state key in the completion step
 * `OAuthClient` doesn't have to know about `channelID`

Please let me know what you think of this approach. I hope I didn't miss anything important this time 😄

Reply:
*Follow-up to my previous comment:*

I added a new commit to fix an error handing bug: https://github.com/mattermost/mattermost-plugin-zoom/pull/150/commits/01e68f9f5cd0b13ec7304cc2ae38ac3f0dc5cad2

But now I realize that I can't set the state in the KV store inside `/zoom start` command handler. The following line always returns an error:
https://github.com/utkuufuk/mattermost-plugin-zoom/blob/01e68f9f5cd0b13ec7304cc2ae38ac3f0dc5cad2/server/command.go#L73

Moreover, calling `appErr.Error()` inside `p.API.LogWarn` crashes the plugin:
```
ERRO[...] RPC call ExecuteCommand to plugin failed.     caller="mlog/log.go:190" error="unexpected EOF" plugin_id=zoom
```

I could use some help on how to debug this. 😅 
Any idea why calling `KVSet` or `appErr.Error()` would fail?

Actually it would be really helpful if I was able to debug the plugin code (e.g. using VS Code) after I deploy it to the server. Is there a guide or readme for debugging MM plugins?